guacamole/src/main/java/org/apache/guacamole/rest/auth/AuthenticationService.java

```diff
@@ -232,11 +286,17 @@ private AuthenticatedUser getAuthenticatedUser(GuacamoleSession existingSession,
         try {
 
             // Re-authenticate user if session exists
-            if (existingSession != null)
-                return updateAuthenticatedUser(existingSession.getAuthenticatedUser(), credentials);
+            if (existingSession != null) {
+                AuthenticatedUser updatedUser = updateAuthenticatedUser(
+                        existingSession.getAuthenticatedUser(), credentials);
+                notifyAuthenticationSuccessListeners(updatedUser, existingSession);
+                return updatedUser;
+            }
 
             // Otherwise, attempt authentication as a new user
-            AuthenticatedUser authenticatedUser = AuthenticationService.this.authenticateUser(credentials);
+            AuthenticatedUser authenticatedUser = authenticateUser(credentials);
```

Comment:
I suspect that my IDE noticed that the explicit `this` wasn't needed and removed it. I haven't looked at the history of this piece of code, but I suspect that it was previously refactored from a state in which defererencing `authenticateUser` in this way was necessary. I'm happy to revert this change if desired.